### ISSUE TITLE
Mismatch in Top property of range (address format like H21:I22 ) when zoom level of sheets are different.

### ISSUE THREAD

**Issue by techindia2010** [OfficeDev/office-js#2068]
# Mismatch in Top property of range (address format like H21:I22 ) when zoom level of sheets are different.

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: PC desktop
* Host [Excel, Word, PowerPoint, etc.]:Excel
* Office version number: Microsoft 365 Apps for enterprise ver 2102
* Operating System: Window 10
* Browser (if using Office on the web): NA
 
## Expected behavior

Top property of range shall not be dependent on zoom level of any sheet. Other property like height, left and width do not have any influence of zoom level of sheet. In fact, if address of range is single cell (H25),  Top property don’t have any influence of zoom level.

## Current behavior

only Top property of range changed if zoom level of sheet changed.

## Steps to reproduce

1.  Load top property of range (address format shall be range area, ie, H21:I22) of two different sheets. One sheet must have different zoom level (Example 200%) from another sheet (Example 100%). 

## Link to live example(s)

1. [Script Lab](https://gist.github.com/techindia2010/d4eaa9a6174351eb3d98bb6063e67b79)

# Provide additional details

1. [Script Lab](https://gist.github.com/techindia2010/d4eaa9a6174351eb3d98bb6063e67b79)

## Context

1.  Not able to insert image at correct location for all sheets at selected range location if zoom level is different.  There is not API to set zoom level of sheet.
## Useful logs

- [ ] Console errors
- [ ] Screenshots
- [ ] Test file (if only happens on a particular file)

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by qinliuMSFT** [comment]
Hi @techindia2010, thank you for logging this issue. The issue is reproducible for us, and has been put on out backlog VSO#5360258, will let you know once we have any updates.

**Comment by honxmsft** [comment]
Hi @techindia2010, we are currently working on the fix. Meanwhile, we found a workaround which possibly can unblock you scenario. 

Instead to use the `range.top`, you can use the `range.height` API to get correct top.
For example, if you want to get the `range.top` of the range `H25:I26`, it's equivalent to get the `range.height` of range `A24:H24`, which won't be affected by active worksheet.

In more general form, you can have such function to get correct top value as a workround:

```ts
async function getRangeTop(range: Excel.Range, context: Excel.RequestContext) {
  // load row index of the range to get top
  range.load(['rowIndex'])
  await context.sync()
  // get the complementary range from 0 to up one cell of the input range
  const complementaryRange = range.worksheet.getRangeByIndexes(0, 0, range.rowIndex, 1)
  // load the height
  complementaryRange.load(['height'])
  await context.sync()
  // the height is the correct top
  return complementaryRange.height
}
```

Hope this can help you. Let me know if you have any question 😃

**Comment by ronald-forscale** [comment]
Hi @honxmsft , it seems the workaround is too good to be true. The loaded top and height properties are independent from the zoom, it is the placement of the shapes that depends on the zoom.

**Comment by donlvMSFT** [comment]
The issue already been fixed due to the bug status, and released after office version: 16.0.14715.34850, please help verify.
I'd like to close this thread, thanks for the discussion!

**Comment by doantumai** [comment]
Hi @honxmsft I'm not having the same issue but do you know how to set the zoom level of the worksheet? I tried this https://learn.microsoft.com/en-us/javascript/api/excel/excel.pagelayout?view=excel-js-preview#excel-excel-pagelayout-zoom-member but the zoom level wasn't changed.
Instead, if I go to Page Layout > Page Setup > Page > Scaling is adjusted to 200%. There is no other API to set zoom level instead of this. Thank you!
